### PR TITLE
Refactor unit more

### PR DESCRIPTION
**Pull request by royfalk**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [x] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues **None**

Purpose:
- What is this pull request trying to do? **Refactor code by removing redundant unit_factory files.**
- What release is this for? **0.7 probably**
- Is there a project or milestone we should apply this to?

### CODE REVIEW

**Comment by royfalk** [comment]
Note that this PR builds on the previous one and is only https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/214/commits/8e0c9399364ef46952c2f0b657d74769e0f71eb6. It's part of my attempt to create smaller, faster PR's.

**Comment by BenjamenMeyer** [comment]
@royfalk once your previous is merged then just do:
```
git checkout master
git fetch upstream
git merge upstream/master
git checkout refactor_unit_more
git merge master
...fix any merge issues...
git push origin refactor_unit_more
```
if all is well and it matches up, then you'll just have a fast-forward merge.

File: engine/src/cmd/asteroid.h [hunk 1]
```diff
@@ -20,7 +19,7 @@ class GameAsteroid : public GameUnit< Asteroid >
                                  const Vector &CumulativeVelocity,
                                  bool ResolveLast,
                                  UnitCollection *uc = NULL );
-protected:
+
 /** Constructor that can only be called by the UnitFactory.
```
Comment (stephengtuggy) [inline engine/src/cmd/asteroid.h:23]: Might want to take this comment out

File: engine/src/cmd/enhancement.h [hunk 2]
```diff
@@ -7,7 +7,7 @@
 class GameEnhancement : public GameUnit< Enhancement >
 {
     friend class UnitFactory;
-protected:
+public:
 /// constructor only to be called by UnitFactory
```
Comment (stephengtuggy) [inline engine/src/cmd/enhancement.h:11]: Might want to remove this comment as well

File: engine/src/cmd/missile.h [hunk 3]
```diff
@@ -78,14 +80,6 @@ class GameMissile : public GameUnit< Missile >
...
                                              uc );
     }
-
-private:
-/// default constructor forbidden
-    GameMissile();
-/// copy constructor forbidden
-    GameMissile( const Missile& );
-/// assignment operator forbidden
-    GameMissile& operator=( const Missile& );
```
Comment (stephengtuggy) [inline engine/src/cmd/missile.h:88]: Hmm. We are no longer forbidding these constructors? I'll have to think about that.

File: engine/src/cmd/unit_factory.cpp [hunk 4]
```diff
@@ -1,238 +0,0 @@
...
-    return tt;
-}
-
-ContinuousTerrain* UnitFactory::createContinuousTerrain( const char *file, Vector scale, float position, Matrix &t )
-{
-    ContinuousTerrain *ct;
-    ct = new ContinuousTerrain( file, scale, position );
-    ct->SetTransformation( t );
-    return ct;
-}
```
Comment (stephengtuggy) [inline engine/src/cmd/unit_factory.cpp:237]: I guess these weren't being used anywhere, were they?

**Comment by stephengtuggy** [comment]
Thanks for the smaller PR, by the way.

File: engine/src/cmd/missile.h (hunk 3 above)
Comment (stephengtuggy) [reply engine/src/cmd/missile.h:88]: @royfalk I suspect that there was a reason why the previous developer(s) forbade these constructors. Can you please put this code back in? Thanks!

File: engine/src/cmd/asteroid.h (hunk 1 above)
Comment (royfalk) [reply engine/src/cmd/asteroid.h]: Done

File: engine/src/cmd/enhancement.h (hunk 2 above)
Comment (royfalk) [reply engine/src/cmd/enhancement.h]: Done

File: engine/src/cmd/missile.h (hunk 3 above)
Comment (royfalk) [reply engine/src/cmd/missile.h]: You are absolutely right. I did not fully grasp the purpose of what they did here. Returned the code for now.

File: engine/src/cmd/unit_factory.cpp (hunk 4 above)
Comment (royfalk) [reply engine/src/cmd/unit_factory.cpp:237]: Sure looks that way. The current state of the code is such that a lot of stuff is either obsolete or in preparation for something the original dev didn't get to do. It's very frustrating and hard to understand, as there's no spec of issues documenting it anymore.

**Comment by stephengtuggy** [comment]
OK, I just ran the PR Validation Tests. Everything looks good. Merging.

**Pull request merged**